juju/juju
# port pr 8317: Adding container-inherit-properties to model-config

## Description of change

container-inherit-properties allows a user to specify cloudinit settings which should be replicated from a machine to a containers which juju has provisioned.

## QA steps

1. Configure MaaS to setup apt settings, mirror or proxy etc.
2. Bootstrap and provision a machine.
3. juju model-config container-inherit-properties="apt-primary"
4. add a container, lxd and kvm should be used.
5. verify the container picked up the apt cloud init properties
6. Try 2 series, one of them trusty, host and container with the same series.
7. Only ca-certs, apt-security, apt-sources and apt-primary allowed.

## Documentation changes

Documentation updates will be needed.

## Bug reference

n/a
